README.md
# Introduction
EVEmu modified to support the latest Incursion client

# Status
- Login works again
- Character selection and creation is working
- Character portraits are next!

# Legal
	------------------------------------------------------------------------------------
	LICENSE:
	------------------------------------------------------------------------------------
	This file is part of EVEmu: EVE Online Server Emulator
	Copyright 2006 - 2011 The EVEmu Team
	For the latest information visit http://evemu.org
	------------------------------------------------------------------------------------
	This program is free software; you can redistribute it and/or modify it under
	the terms of the GNU Lesser General Public License as published by the Free Software
	Foundation; either version 2 of the License, or (at your option) any later
	version.

	This program is distributed in the hope that it will be useful, but WITHOUT
	ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
	FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.

	You should have received a copy of the GNU Lesser General Public License along with
	this program; if not, write to the Free Software Foundation, Inc., 59 Temple
	Place - Suite 330, Boston, MA 02111-1307, USA, or go to
	http://www.gnu.org/copyleft/lesser.txt.
	------------------------------------------------------------------------------------
	

doc/ChangeLog
 
*** 22/07/2008 ***
- Added autotools

*** 03/05/2008 ***
- Eliminate invBlueprints table by pulling data from other existing tables. (Bloody.Rabbit)
- Implemented bubble manager to handle in-space items which are not visible system wide.
- Partial support for boarding an empty ship in space.
- Properly undock to the station's dock location (firefoxpdm)
- Properly use ships position in space from the DB.
- Store ship position in space when logging out.
- Improved support for fitting charges with weapons.
- Improved NPC AI to make them consistently follow, orbit and attack.
- Hybrid and Laser modules consistently fire on NPCs now.
- Implemented damage code including basic messages.
- Implemented NPC death logic. 
- Implement code to support asteroids in space.
- Add new command to spawn an asteroid: /roid (typeID) (radius)
- Implemented graphics for mining lasers (they do not yet actually mine though)
- Load up space junk (roids, items, etc) from DB ay system boot.
- Implement runtime memory error exception handler to help prevent one bad action from crashing the whole server.

*** 02/15/2008 ***
- Load tutorial information from DB (Bloody.Rabbit)

*** 02/03/2008 ***
- Celestial Statistics support (firefoxpdm)

*** 11/24/2007 ***
- Reworked bind ID storage code (firefoxpdm)
- More coporation cleanup (firefoxpdm)
- Fix issue with the wallet (firefoxpdm)
- Fixed "show info" tab for DoGetStation (firefoxpdm)
- Fixed station backgrounds (firefoxpdm)
- Implemented people and places lookup (firefoxpdm)
- Fix for owner notes (GuristaSyndicate)
alter table chrOwnerNote CHANGE content note TEXT NOT NULL;

*** Release 229 ***

*** 10/20/2007 ***
- Intial market implementation:
  - Major work on special market packet encodings.
  - Browse market, see pending orders.
  - Initial price history implementation.
  - Place sell orders (builds the market)
  - Place buy orders
  - Place immediate sell orders
  - Place immediate buy orders
  - you prolly need to log out&in before seeing new orders right now.
- Major destiny (space) work:
  - Much improved event broadcasting to allow players to 
    observe the actions of another player.
  - New module manager to handle different weapon types
  - Improved timer support for module activation/deactivation.
  - Can now activate weapons and generate fake hits on targets.
  - Initial infrastructure for damage propigation.
- A bit more corporation creation work. (firefoxpdm)
- Initial NPC AI work. Very very simple right now.
- Added /spawn (typeID) command to spawn an NPC.
- Started reworking chat channels a bit. It is prolly more broken
  than it was previously right now, but should improve over time.
- Start rework on chat channels to support dynamic channels.
- Initial XML packet support for none-value duality handling.
- Added queue for destiny updates to properly batch them up.
Update SQL:
ALTER TABLE market_orders DROP `isBuy`;
ALTER TABLE market_orders CHANGE `bid` `bid` tinyint(3) unsigned NOT NULL default '0';
ALTER TABLE market_orders CHANGE `issued` `issued` bigint(20) unsigned NOT NULL default '0';
ALTER TABLE market_orders CHANGE `issued` `issued` bigint(20) unsigned NOT NULL default '0';
ALTER TABLE market_transactions CHANGE `transactionDateTime` `transactionDateTime` bigint(20) unsigned NOT NULL default '0';
ALTER TABLE market_transactions ADD `regionID` int(10) unsigned NOT NULL default '0';
New Table: billsPayable
New Table: market_history_old

*** 07/25/2007 ***
- Major revelation with destiny updates, fixes 'jumpy' movement in space.
- Much better understanding of the binary destiny update structure.
- Significant work on the destiny manager. Server now tracks client movement for 
  warp, goto direction, and follow movement modes (orbit is the only major mode pending).
- Some initial drone launching code (not working yet)
- A bunch of makefile cleanup.
- Added support for the <files><cache>.. element in config file to cache the cache files
- Implemented a keepalive ping to hold idle sessions open.
- Added support for server originated messsages: error, infomodal, and notify
- Made persistent attributes auto-store.
- Fix for crash when jumping between systems.
- Owner Note implemented. (firefoxpdm)
- Corp channel creation and logo fixes (firefoxpdm)
- A bunch of other corporation creation stuff (firefoxpdm)

*** 5/01/2007 - 182 ***
- Initial targeting system
- Some support for training new skills from a skill book. It may work some times.
- added new command /search which searches items for summoning purposes
- added functionality to support the server sending you evemails (like for long search results)
- Initial standing history support (firefoxpdm)
- Initial corp creation code (firefoxpdm)
- Reworked the database priming technique.
- Merged the agtAgents2 table into agtAgents
- A few fixes for static owner querying.
- Initial ability to equip and activate modules (may not persist properly)

SQL Changes:
new table: chrStandingChanges
ALTER TABLE `corporation` CHANGE `corporationID` `corporationID` INT( 10 ) UNSIGNED NOT NULL AUTO_INCREMENT;
ALTER TABLE `corporation`  AUTO_INCREMENT =2000001;

*** 4/22/2007 - 169 ***
Major increase in error checking of generated xmlp objects
New FastEncode() operation on generated xmlp objects to reduce copying
A new mode of element operations in xmlp (non-pointer)
Major reworking of all item stuff to use the new inventory system


alter table entity AUTO_INCREMENT=140000000;
alter table dgmAttributeTypes CHANGE attributeID attributeID INTEGER UNSIGNED;
new table entity_attributes

*** 04/15/2007 ***
- Implemented initial employment history support (firefoxpdm)
- Implemented Character Note (LSMoura)
- Got evetool building on win32, still needs some work though.
- Added `unmarsahal` command to evetool to decode strings from logserver.

src/eve-tool/main.cpp
/*
    ------------------------------------------------------------------------------------
    LICENSE:
    ------------------------------------------------------------------------------------
    This file is part of EVEmu: EVE Online Server Emulator
    Copyright 2006 - 2011 The EVEmu Team
    For the latest information visit http://evemu.org
    ------------------------------------------------------------------------------------
    This program is free software; you can redistribute it and/or modify it under
    the terms of the GNU Lesser General Public License as published by the Free Software
    Foundation; either version 2 of the License, or (at your option) any later
    version.

    This program is distributed in the hope that it will be useful, but WITHOUT
    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
    FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.

    You should have received a copy of the GNU Lesser General Public License along with
    this program; if not, write to the Free Software Foundation, Inc., 59 Temple
    Place - Suite 330, Boston, MA 02111-1307, USA, or go to
    http://www.gnu.org/copyleft/lesser.txt.
    ------------------------------------------------------------------------------------
    Author:        Zhur
*/

#include "EVEToolPCH.h"

#include "main.h"
#include "Commands.h"

const char* const CACHE_DIR =         EVEMU_ROOT_DIR"cache";
const char* const LOG_FILE =          EVEMU_ROOT_DIR"log/eve-tool.log";
const char* const LOG_SETTINGS_FILE = EVEMU_ROOT_DIR"etc/log.ini";

int main( int argc, char* argv[] )
{
#if defined( MSVC ) && !defined( NDEBUG )
    // Under Visual Studio setup memory leak detection
    _CrtSetDbgFlag( _CRTDBG_LEAK_CHECK_DF | _CrtSetDbgFlag( _CRTDBG_REPORT_FLAG ) );
#endif /* defined( MSVC ) && !defined( NDEBUG ) */

    // Load server log settings ( will be removed )
    if( !load_log_settings( LOG_SETTINGS_FILE ) )
        sLog.Warning( "init", "Unable to read %s (this file is optional)", LOG_SETTINGS_FILE );
    else
        sLog.Success( "init", "Log settings loaded from %s", LOG_SETTINGS_FILE );

    if( !log_open_logfile( LOG_FILE ) )
        sLog.Warning( "init", "Unable to open log file '%s', only logging to the screen now.", LOG_FILE );
    else
        sLog.Success( "init", "Opened log file %s", LOG_FILE );

    //skip first argument (launch path), we don't need it
    --argc;
    ++argv;

    for(; argc > 0; --argc, ++argv)
        ProcessFile( *argv );

    ProcessFile( stdin );

    sLog.Log( "shutdown", "Exiting." );

    return 0;
}

void ProcessString( const char* str )
{
    ProcessCommand( Seperator( str ) );
}

void ProcessString( const std::string& str )
{
    ProcessString( str.c_str() );
}

void ProcessFile( FILE* file )
{
    std::string cmd;

    while( true )
    {
        char input = fgetc( file );

        if( '\n' == input || EOF == input )
        {
            if( !cmd.empty() )
            {
                ProcessString( cmd );
                cmd.clear();
            }

            if( EOF == input )
                break;
        }
        else
            cmd += input;
    }
}

void ProcessFile( const char* filename )
{
    ProcessFile( std::string( filename ) );
}

void ProcessFile( const std::string& filename )
{
    FILE* file = fopen( filename.c_str(), "r" );
    if( NULL == file )
    {
        sLog.Error( "input", "Unable to open script '%s'.", filename.c_str() );
        return;
    }

    sLog.Log( "input", "Queuing commands from script '%s'.", filename.c_str() );
    ProcessFile( file );

    if( feof( file ) )
        sLog.Success( "input", "Load of script '%s' successfully completed.", filename.c_str() );
    else
        sLog.Error( "input", "Error occured while reading script '%s'.", filename.c_str() );

    fclose( file );
}




